pkg/profiler/rlimit.go

```diff
@@ -16,22 +16,35 @@ package profiler
 
 import (
 	"fmt"
+	"sync"
 	"syscall"
 
 	"golang.org/x/sys/unix"
 )
 
+var rlimitMu sync.Mutex
+
 // BumpMemlock increases the current memlock limit to a value more reasonable for the profiler's needs.
 func BumpMemlock(cur, max uint64) (syscall.Rlimit, error) {
 	// TODO(kakkoyun): https://github.com/cilium/ebpf/blob/v0.8.1/rlimit/rlimit.go
+	if cur == 0 {
+		cur = unix.RLIM_INFINITY
+	}
+	if max == 0 {
+		max = unix.RLIM_INFINITY
+	}
 	rLimit := syscall.Rlimit{
 		Cur: cur,
 		Max: max,
 	}
+
+	rlimitMu.Lock()
```

Comment:
Why is this needed?

We don't call it concurrently no? Either way, it seems like it's thread-safe:

>         │getrlimit(), setrlimit(), prlimit()   │ Thread safety │ MT-Safe │

https://man7.org/linux/man-pages/man2/getrlimit.2.html

Reply:
It's primarily future envy. We will have more than one profiler eventually, and this API is designed to be called from the profilers. Happy to get rid of it.